Path-Check/covid-safe-paths
# Runtime Feature Flags

#### Description:

Implements runtime feature flagging. Default values for the flags are set from the respective build time env file.

Also creates a `<Switch />` component.

#### Screenshots:

![Simulator Screen Shot - iPhone 11 - 2020-05-03 at 17 40 53](https://user-images.githubusercontent.com/20157849/80928094-df8e1500-8d67-11ea-9e75-2e01d283625c.png)

#### How to test:

- _Toggles hide the feature as expected_
  - Verify that toggling `Exposure mode` and `google_imports` toggles the respective features in the UI. 
  - Toggle `better_location_status_checks` and verify that `<LocationTracking />` is displayed instead (this uses the `fallback` prop)
- _Toggles persist in async storage_
  - Toggle `google_imports` to off (non-default state). Restart the app. The value should have persisted, and the toggle should still be off.
- _`.env` is the source of truth for which toggles are available_
  - Add a new toggle to `.env`, e.g. `flag_toggle_test`. Run `npx react-native run-android` to rebuild with the new flag. Open the app and toggle the new flag so that it persists to storage. Remove the flag from `.env` and then re-run `npx react-native run-android`. The flag should not be displayed on the toggle screen.
- _The feature flags screen only displays in dev mode_
  - Build a release on either platform and confirm that the `Feature Flags` nav button does not display.
